src/Symfony/Component/Messenger/Exception/EnvelopeAwareExceptionInterface.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Exception;
+
+use Symfony\Component\Messenger\Envelope;
+
+interface EnvelopeAwareExceptionInterface
```

Comment:
Symfony follows [semver](https://semver.org) which doesn't allow expanding the public API in patch releases. As this is a bugfix, the interface + trait should be reverted or at least be marked as internal. They can be introduced separately as an improvement on the development branch (currently 7.1).

Reply:
Thanks, marked both as internal. I think it'd be great to expose the interface as public in 7.1 to allow proper behavior with user-space middlewares too.